File: Pascal_Compiler/test_file.pas
PROGRAM TestComplet;

{ 
  Ceci est un commentaire multiligne.
  Le lexer doit ignorer tout ça.
}

VAR
    compteur, resultat : INTEGER;
    nom_variable_long : INTEGER;

BEGIN
    { Test des affectations et nombres }
    compteur := 10;
    resultat := 0;

    { Test des conditions et operateurs logiques }
    IF compteur > 5 THEN
    BEGIN
        resultat := compteur + 2;
    END
    ELSE
    BEGIN
        resultat := compteur - 1;
    END;

    { Test de la boucle et operateurs math }
    WHILE compteur < 20 DO
    BEGIN
        compteur := compteur + 1;
        resultat := resultat * 2 / (5 - 3);
        WRITE(resultat); { Test appel fonction }
    END;

    { Test de la fin du programme }
END.

File: Pascal_Compiler/test_file.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    // Variables
    int compteur;
    int resultat;
    int nom_variable_long;

    // Programme Principal
    compteur = 10;
    resultat = 0;
    if ((compteur > 5)) {
    resultat = (compteur + 2);
    }
    else {
    resultat = (compteur - 1);
    }
    while ((compteur < 20)) {
    compteur = (compteur + 1);
    resultat = ((resultat * 2) / (5 - 3));
    printf("%d\n", resultat);
    }

    return 0;
}
